### PR TITLE
Merge unecessary subprojects and remove CLI argument parsing

### PR DESCRIPTION
**Pull request by showermat**

### CODE REVIEW

File: cli/src/main/scala/com/ibm/sparktc/sparkbench/cli/CLIKickoff.scala [hunk 1]
```diff
@@ -1,37 +1,15 @@
...
   override def main(args: Array[String]): Unit = {
-//    println(s"hello args! ${args.toSeq}")
-    val sArgs = new ScallopArgs(args)
-
-    sArgs.subcommand match {
-      case Some(sArgs.datagen) => startDataGen(sArgs)
-      case Some(sArgs.workload) => startWorkload(sArgs)
-      case None => useConfFile(sArgs)
-      case _ => new Exception(s"Unrecognized subcommand.\n${sArgs.printHelp()}")
+    args.length match {
```
Comment (cin) [inline cli/src/main/scala/com/ibm/sparktc/sparkbench/cli/CLIKickoff.scala:9]: I'd prefer `args.headOption` here.

File: cli/src/test/scala/com/ibm/sparktc/sparkbench/OutputTest.scala [hunk 2]
```diff
@@ -19,18 +19,13 @@ class OutputTest extends FlatSpec with Matchers with BeforeAndAfterAll with Capt
   }
 
   "Specifying Console output" should "work" in {
-    val (out) = captureOutput(CLIKickoff.main(
-      Array("bin/spark-bench.sh", "workload", "kmeans", "-i", dataStuff.kmeansFile, "-o", "console", "-n", "5", "-k", "2"))
-    )
+    val (out) = captureOutput(CLIKickoff.main(Array(getClass.getResource("/etc/testConfFile3.conf").getPath)))
```
Comment (cin) [inline cli/src/test/scala/com/ibm/sparktc/sparkbench/OutputTest.scala]: I see `(out)` was like this before too but why?

**Review by cin — CHANGES_REQUESTED** [review]
So much red...<3 it

File: cli/src/test/scala/com/ibm/sparktc/sparkbench/OutputTest.scala (hunk 2 above)
Comment (ecurtin) [reply cli/src/test/scala/com/ibm/sparktc/sparkbench/OutputTest.scala]: I can answer that. It was like that I was using a method from Capturing that output a tuple of stuff, and then I switched to captureOutput which only captures stdout.

File: cli/src/test/scala/com/ibm/sparktc/sparkbench/cli/HelpersTest.scala [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package com.ibm.sparktc.sparkbench.cli
+
+import com.ibm.sparktc.sparkbench.workload.Suite
+import com.typesafe.config.ConfigException
+import org.scalatest.{FlatSpec, Matchers}
+
+class HelpersTest extends FlatSpec with Matchers {
```
Comment (ecurtin) [inline cli/src/test/scala/com/ibm/sparktc/sparkbench/cli/HelpersTest.scala:7]: This test is appreciated!

File: cli/src/main/scala/com/ibm/sparktc/sparkbench/workload/ml/MLWorkloadResults.scala [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package com.ibm.sparktc.sparkbench.workload.ml
+
+case class MLWorkloadResults(
```
Comment (ecurtin) [inline cli/src/main/scala/com/ibm/sparktc/sparkbench/workload/ml/MLWorkloadResults.scala]: I'm pretty sure this file can get deleted, this code isn't getting touched anywhere. That may be taken care of in my latest changes in my branch though.

File: cli/src/test/scala/com/ibm/sparktc/sparkbench/workload/ml/KMeansWorkloadTest.scala [hunk 5]
```diff
@@ -0,0 +1,73 @@
+package com.ibm.sparktc.sparkbench.workload.ml
+
+import java.io.File
+
+import com.holdenkarau.spark.testing.{DataFrameSuiteBase, Utils}
```
Comment (ecurtin) [inline cli/src/test/scala/com/ibm/sparktc/sparkbench/workload/ml/KMeansWorkloadTest.scala]: DataFrameSuiteBase should be a dead import.

**Review by ecurtin — COMMENTED** [review]
Overall this looks really good. I appreciate your attention to detail!

File: cli/src/main/scala/com/ibm/sparktc/sparkbench/workload/ml/MLWorkloadResults.scala (hunk 4 above)
Comment (showermat) [reply cli/src/main/scala/com/ibm/sparktc/sparkbench/workload/ml/MLWorkloadResults.scala]: Right you are.  I'm deleting it as well, and it should merge fine if we both delete.

File: cli/src/main/scala/com/ibm/sparktc/sparkbench/cli/CLIKickoff.scala (hunk 1 above)
Comment (showermat) [reply cli/src/main/scala/com/ibm/sparktc/sparkbench/cli/CLIKickoff.scala:9]: If I do that, what's the appropriate way to reject more than one argument?

File: cli/src/test/scala/com/ibm/sparktc/sparkbench/OutputTest.scala (hunk 2 above)
Comment (showermat) [reply cli/src/test/scala/com/ibm/sparktc/sparkbench/OutputTest.scala]: Er, it looks like you're still using function `captureOutput` in trait `Capturing` that returns a `(String, String)`.  I'm not sure what's going on here syntactically, but I converted `(out)` to `(out, _)` and it still compiles and tests fine.  Does that make any sense?

**Pull request merged**